### PR TITLE
Ease .btn alignment by switching to inline-flex

### LINKED ISSUE #30837: Wrong vertical align in button links
I saw that the button links doesn't obey to `vertical-align` style. It keep being to top, not in the middle. [Here](https://jsfiddle.net/ebj9L6vx/) is visible this problem.

### PR DESCRIPTION
**Pull request by ffoodd**
Fixes #30837 

Simple PR to try this out. Should probably be backported to v4 if merged.

### CODE REVIEW

File: scss/_buttons.scss [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+  display: inline-flex;
+  align-items: center;
   font-family: $btn-font-family;
   font-weight: $btn-font-weight;
   line-height: $btn-line-height;
   color: $body-color;
   text-align: center;
   text-decoration: if($link-decoration == none, null, none);
   white-space: $btn-white-space;
-  vertical-align: middle;
```
Comment (MartijnCuppens) [inline scss/_buttons.scss:14]: `vertical-align` applies to aligning the button to its parent, not aligning its content.

**Comment by MartijnCuppens** [comment]
> Should probably be backported to v4 if merged.

This will probably cause lot's of breaking changes. for example see https://codepen.io/MartijnCuppens/pen/LYpwjXy

**Comment by ffoodd** [comment]
So does it make sense, though?

I think easier buttons alignment is very handy since it's a very common use case—however I'm not sure about the side-effects compared to v4.

**Comment by mdo** [comment]
Inclined to hold on this for the side effects and what not. Let's keep things closer to v4 where we can and see what paradigms we can really break down and reinvent in v6 :).

**Comment by mdo** [comment]
Going to close this out (would be changing to `main` branch anyway post alpha 1) for now. Can always dig it back up and restore if we need to. <3

**Pull request closed without merging**